HOK.Core/HOK.Core/BackgroundTasks/Rules.cs

```diff
@@ -0,0 +1,161 @@
+                            SpecTypeId.Boolean.YesNo
+                        );
+                        // Set the default value to off/false
+                        hokPrintSetParam.SetValue(new IntegerParameterValue(1));
+                    }
+                    else
+                    {
+                        tr.Start();
+                        GlobalParameter hokPrintSetParam = GlobalParameter.Create(
+                            doc,
+                            "Add ALL Sheets to HOK Print Set",
+                            SpecTypeId.Boolean.YesNo
+                        );
+                        // Set the default value to off/false
+                        hokPrintSetParam.SetValue(new IntegerParameterValue(1));
+                        tr.Commit();
+                    }
+                }
+            }
+            hokPrintSetParamId = GlobalParametersManager.FindByName(
+                doc,
+                "Add ALL Sheets to HOK Print Set"
+            );
+            var hokPrintSetParameter = doc.GetElement(hokPrintSetParamId) as GlobalParameter;
+            if ((hokPrintSetParameter.GetValue() as IntegerParameterValue).Value == 1)
+            {
+                goto AddAllSheets;
+            }
+            else
+            {
+                return;
+            }
+
+        AddAllSheets:
+            PrintManager pm = doc.PrintManager;
+
+            // Ensure that the print manager setting is set to selected views
+            if (pm.PrintRange != PrintRange.Select)
+            {
+                pm.PrintRange = PrintRange.Select;
+            }
+
+            ViewSheetSetting existingVSS = pm.ViewSheetSetting;
+            // Save the previously selected viewsheet set so we can set it back to the original at the end
+            var existingViewSheetSet = existingVSS.CurrentViewSheetSet;
+
+            var hokSheetSet = new FilteredElementCollector(doc)
+                .OfClass(typeof(ViewSheetSet))
+                .Cast<ViewSheetSet>();
+
+            if (hokSheetSet.Select(ss => ss.Name).Contains("_HOK Automated Print Set"))
+                existingVSS.CurrentViewSheetSet = hokSheetSet.First(
+                    ss => ss.Name == "_HOK Automated Print Set"
+                );
+
+            List<ElementId> sheetsInModel = new FilteredElementCollector(doc)
+                .OfClass(typeof(ViewSheet))
+                .ToElements()
+                .Cast<ViewSheet>()
+                .Select(s => s.Id)
+                .ToList();
+
+            // Add the excluded sheets to the sheet set
+            ViewSet newVSS = new ViewSet();
+            foreach (var viewId in sheetsInModel)
+            {
+                View currView = (View)doc.GetElement(viewId);
+                if (!newVSS.Contains(currView))
+                {
+                    newVSS.Insert(currView);
+                }
+                else
+                {
+                    continue;
+                }
```

Comment:
This is redundant, since there's no code below this statement it will already just go ahead and continue, so we can remove the `else` section for clarity.